pkg/system/phase2_creating.go

```diff
@@ -171,6 +176,101 @@ func (r *Reconciler) ReconcilePhaseCreatingForMainClusters() error {
 	return nil
 }
 
+func (r *Reconciler) UpgradeDbPostgres16() error {
+	var err error = nil
+
+	// Check the desired image, if it is not as same as db image to upgrade,
+	// don't consider for upgrade
+	// Note: Intention of the below call is to fetch the desired image from Noobaa CR or env var
+	// Passing empty Arg "" to GetDesiredDBImage() in the below case
+	desiredImage := GetDesiredDBImage(r.NooBaa, "")
+	if !strings.Contains(desiredImage, "postgresql-16") {
```

Comment:
what is the name of the DS image? can we rely on this comparison?

As an alternative for the image name comparison, you can first compare the desired image to the one set in the DB STS:
- if the images are the same, nothing to do
- if they differ AND the current major version is not 16 yet, start the upgrade and modify the upgrade phase

Reply:
upstream: quay.io/sclorg/postgresql-16-c9s 
downstream: registry.redhat.io/rhel9/postgresql-16

Reply:
`isPostgresUpgradeReqd()` does 
```

    if the images are the same, nothing to do
    if they differ AND the current major version is not 16 yet, start the upgrade and modify the upgrade phase

```

This check is done in 2 cases
1. case "": This is for the 1st time when we upgrade to the new build, the `PostgresUpdatePhase` is empty
2. case "UpgradePhaseNone": This check is done everytime until the Postgres is upgrade to 16. As soon as desired image is `postgres-16`, we upgrade to `postgres-16` and set the status to upgrade finished.